ktorm-support-mysql/src/main/kotlin/me/liuwj/ktorm/support/mysql/Functions.kt

```diff
@@ -167,6 +170,30 @@ fun <T : Any> ColumnDeclaring<T>.ifNull(right: T?): FunctionExpression<T> {
     return this.ifNull(wrapArgument(right))
 }
 
+/**
+ * MySQL sum & if combination, translated to `sum( if(column condition, ifTrue, ifFalse) )`.
+ */
+fun sumIf(condition: BinaryExpression<*>, ifTrue: Int, ifFalse: Int): AggregateExpression<Int> {
```

Comment:
Would it be better to have an `IF` function instead of `sumIf`: 

```kotlin
fun <T : Any> IF(
    condition: ColumnDeclaring<Boolean>,
    then: ColumnDeclaring<T>,
    otherwise: ColumnDeclaring<T>
): FunctionExpression<T> {
    // if(condition, then, otherwise)
    return FunctionExpression(
        functionName = "if",
        arguments = listOf(condition, then, otherwise).map { it.asExpression() },
        sqlType = then.sqlType
    )
}
```

Usage: 

```kotlin
val query = database
    .from(Employees)
    .select(sum(IF(Employees.salary greaterEq 100L, then = 1, otherwise = 0)))
    .where { Employees.salary greaterEq 100L }
```

I think it would be more flexible. What's your opinion?

Reply:
@vincentlauvlwj I agree to have a `if` function is better.
If this happen, we can write other query like `count( if ( x, y, z) )` .
However, perhaps the word `if` is reserved for Kotlin compiler. To escape them, we might able to use backquote [Escape reserved words in an import](https://stackoverflow.com/questions/36323623/escape-reserved-words-in-an-import).

Reply:
Maybe we can name the function in upper case, like `IF`.